### PR TITLE
Require users to write readme in English

### LINKED ISSUE #1009: Require users to write readme.txt in English
We are on the Plugins Team and require users to write the readme in English. Depending on the method used, we may classify it as either an ERROR or a WARNING.

### PR DESCRIPTION
**Pull request by davidperezgar**
Implements language detection for `readme.txt` files to ensure compliance with WordPress.org requirement that readmes be written in standard English.

## Implementation

Uses `patrickschur/language-detection` library to detect language with enhanced preprocessing:

### Enhanced Detection Features
- ✅ Strips HTML tags, code snippets, URLs, and emails before detection
- ✅ Minimum content length (30 chars) to avoid false positives on short text
- ✅ Confidence thresholds for more accurate results
- ✅ Handles technical content without false positives

### Check Locations & Severity
- **Short Description** and **Full Description**: ERROR (severity 7) - must be English

## Testing

Added comprehensive test coverage:
- Fixed existing test case to properly check errors vs warnings
- Added test for English content (positive case)
- Added test for edge cases (code, URLs, technical terms)
- Created dedicated `Language_Utils` trait tests with 14 test cases
- Created test plugin for edge case validation

All tests pass linting and PHPStan checks.

## Example Output

For non-English content:
```
ERROR: The readme description contains unofficial language. It must be written in standard English.
```

## Edge Cases Handled

✅ Code snippets and technical documentation  
✅ URLs and email addresses  
✅ Short descriptions (< 30 chars get benefit of doubt)  
✅ HTML formatting  
✅ Mixed technical/natural language content  
✅ Common WordPress/plugin terminology  

Fixes #1009

### CODE REVIEW

File: includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php [hunk 1]
```diff
@@ -904,6 +909,49 @@ private function check_requires_headers( Check_Result $result, string $readme_fi
...
+	 * @param string                 $readme_file Readme file.
+	 * @param DotorgParser|PCPParser $parser      The Parser object.
+	 */
+	private function check_language( Check_Result $result, string $readme_file, $parser ) {
+		$check_language = $this->is_on_official_language( $parser->short_description );
+
+		if ( ! $check_language ) {
+			$this->add_result_error_for_file(
+				$result,
+				__( 'The README description contains unofficial language. It must be written in standard English.', 'plugin-check' ),
```
Comment (ernilambar) [inline includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php]: "README" should be simply "readme" I believe.

File: tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php [hunk 2]
```diff
@@ -370,8 +370,8 @@ public function test_run_with_errors_multiple_parser_warnings() {
 		$this->assertNotEmpty( $warnings );
 		$this->assertArrayHasKey( 'readme.txt', $warnings );
 		$this->assertSame( 7, $check_result->get_warning_count() );
-		$this->assertEmpty( $errors );
-		$this->assertSame( 0, $check_result->get_error_count() );
+		$this->assertNotEmpty( $errors );
```
Comment (ernilambar) [inline tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php]: We should improve our test and make sure any addition of new check does not break the existing checks.

**Comment by ernilambar** [comment]
**Suggestions:**

- We should also add tests like we are doing for other traits like https://github.com/WordPress/plugin-check/blob/trunk/tests/phpunit/tests/Traits/URL_Utils_Tests.php
- There is one quirk of short description. Parser gives trimmed short description if there is no actual short description. In that case ERROR of short description is redundant. Its basically checking content of main description in this case. It'd better we handle such case also.

File: includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php (hunk 1 above)
Comment (davidperezgar) [reply includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php]: ok

File: tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php (hunk 2 above)
Comment (davidperezgar) [reply tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php]: ok

**Review by frantorres — APPROVED** [review]

File: includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php [hunk 3]
```diff
@@ -904,6 +910,49 @@ private function check_requires_headers( Check_Result $result, string $readme_fi
 		}
 	}
 
+	/**
+	 * Checks the readme file for official language.
+	 *
+	 * @since 1.0.0
```
Comment (ernilambar) [inline includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php]:
```suggestion
	 * @since 1.8.0
```

File: tests/phpunit/testdata/plugins/test-plugin-plugin-readme-errors-language/readme.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (ernilambar) [inline tests/phpunit/testdata/plugins/test-plugin-plugin-readme-errors-language/readme.txt]: Remove blank line in the initial.

File: tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php [hunk 5]
```diff
@@ -367,23 +367,27 @@ public function test_run_with_errors_multiple_parser_warnings() {
...
+		$this->assertContains( 'readme_parser_warnings_too_many_tags', $warning_codes, 'Should have too many tags warning' );
+		$this->assertContains( 'readme_parser_warnings_requires_header_ignored', $warning_codes, 'Should have requires header ignored warning' );
+		$this->assertContains( 'readme_parser_warnings_tested_header_ignored', $warning_codes, 'Should have tested header ignored warning' );
+		$this->assertContains( 'readme_parser_warnings_requires_php_header_ignored', $warning_codes, 'Should have requires PHP header ignored warning' );
+		$this->assertContains( 'readme_parser_warnings_trimmed_short_description', $warning_codes, 'Should have trimmed short description warning' );
+		$this->assertContains( 'readme_parser_warnings_trimmed_section_changelog', $warning_codes, 'Should have trimmed changelog warning' );
+
+		// Note: This test focuses on parser warnings. Any additional errors from other checks
+		// (like language detection, mismatched headers, etc.) will not cause this test to fail,
+		// making it resilient to new checks being added in the future.
```
Comment (ernilambar) [inline tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php]:
```suggestion
		// Making it resilient to new checks being added in the future.
```

**Review by ernilambar — APPROVED** [review]

File: includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php [hunk 6]
```diff
@@ -904,6 +910,49 @@ private function check_requires_headers( Check_Result $result, string $readme_fi
...
+			if ( ! $check_language ) {
+				$this->add_result_error_for_file(
+					$result,
+					__( 'The readme description contains unofficial language. It must be written in standard English.', 'plugin-check' ),
+					'readme_description_non_official_language',
+					$readme_file,
+					0,
+					0,
+					'https://make.wordpress.org/plugins/2025/07/28/requiring-the-readme-to-be-written-in-english/',
+					7
```
Comment (ernilambar) [inline includes/Checker/Checks/Plugin_Repo/Plugin_Readme_Check.php:950]: Update severity.

File: includes/Traits/Language_Utils.php [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+ */
+
+namespace WordPress\Plugin_Check\Traits;
+
+use LanguageDetection\Language;
+
+/**
+ * Trait for language utilities.
+ *
+ * @since 1.7.0
```
Comment (ernilambar) [inline includes/Traits/Language_Utils.php]:
```suggestion
 * @since 1.8.0
```

File: includes/Traits/Language_Utils.php [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+ */
+trait Language_Utils {
+
+	/**
+	 * Checks if the content is in an official WordPress language.
+	 *
+	 * @since 1.6.0
```
Comment (ernilambar) [inline includes/Traits/Language_Utils.php]:
```suggestion
	 * @since 1.8.0
```
